flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -35,6 +40,112 @@ def test_dataset_creation(self, cluster):
         """
         return create_dataset(self, cluster)
 
+    def _get_package_source(self, default_version=None):
+        """
+        Get the package source for the flocker version under test from
+        environment variables.
+
+        :param unicode default_version: The version of flocker to use
+            if not specified in environment variables.
+
+        :return: A ``PackageSource`` that can be used to install the version of
+            flocker under test.
+        """
+        env_vars = ['FLOCKER_ACCEPTANCE_PACKAGE_BRANCH',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_VERSION',
+                    'FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER']
+        missing_vars = list(x for x in env_vars if x not in os.environ)
+        if missing_vars:
+            raise SkipTest(
+                'Missing environment variables for upgrade test: %s.' %
+                ', '.join(missing_vars))
+        version = (os.environ.get('FLOCKER_ACCEPTANCE_PACKAGE_VERSION') or
+                   default_version)
+        return PackageSource(
+            version=version,
+            branch=os.environ.get('FLOCKER_ACCEPTANCE_PACKAGE_BRANCH'),
+            build_server=os.environ['FLOCKER_ACCEPTANCE_PACKAGE_BUILD_SERVER'])
```

Comment:
Hrm... actually I've got a guard above verifying that they both exist. Agreed that they should both be the same though.

Given that I've got the check above, I think I want the world to blow up (`KeyError` will suffice) if they actually don't exist, as that would indicate that my above checking is not working.